File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/TaskFinishOrOpenCommand.cs
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.TaskManagement.Dispatchs.Commands
{
    /// <summary>
    /// 完工/执行按钮
    /// </summary>
    public class TaskFinishOrOpenCommand : ViewCommand<List<double>>
    {
        protected override object Excute(List<double> args, string scope)
        {
            RT.Service.Resolve<DispatchController>().FinishOrOpenTasks(args);
            return true;
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/TaskFinishOrOpenCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Dispatchs.Commands.TaskFinishOrOpenCommand', {
    meta: { text: "完工/打开", group: "edit", iconCls: "icon-ArrowLeftRight icon-blue" },
    canExecute: function (view) {
        if (view && view.getSelection().length > 0 && view.getSelection().all(p => p.getTaskStatus() == 30 || p.getTaskStatus() == 60))
            return true;
        return false;
    },
    execute: function (view, source) {
        var me = this;
        view.execute({
            data: view.getSelectionIds(),
            success: function (res) {
                if (res.Success) {
                    view.reloadData();
                }
            }
        })
    }
});